lib/network.go

```diff
@@ -5599,6 +5631,100 @@ func (tsl *TransactionSpendingLimit) ToMetamaskString(params *DeSoParams) string
+			stakeLimitStr = append(stakeLimitStr, opString)
+		}
+		// Ensure deterministic ordering of the transaction count limit strings by doing a lexicographical sort.
+		sortStringsAndAddToLimitStr(stakeLimitStr)
+		indentationCounter--
+	}
+
+	// UnstakeLimitMap
+	if len(tsl.UnstakeLimitMap) > 0 {
+		var unstakeLimitStr []string
+		str += _indt(indentationCounter) + "Unstaking Restrictions:\n"
+		indentationCounter++
+		for limitKey, limit := range tsl.UnstakeLimitMap {
+			opString := _indt(indentationCounter) + "[\n"
+
+			indentationCounter++
+			// ValidatorPKID
+			validatorPublicKeyBase58Check := "Any"
+			if !limitKey.ValidatorPKID.Eq(&ZeroPKID) {
+				validatorPublicKeyBase58Check = Base58CheckEncode(limitKey.ValidatorPKID.ToBytes(), false, params)
+			}
+			opString += _indt(indentationCounter) + "Validator PKID: " + validatorPublicKeyBase58Check + "\n"
+			// StakerPKID
+			stakerPublicKeyBase58Check := Base58CheckEncode(limitKey.StakerPKID.ToBytes(), false, params)
+			opString += _indt(indentationCounter) + "Staker PKID: " + stakerPublicKeyBase58Check + "\n"
+			// UnstakeLimit
+			unstakeLimitDESO := float64(limit) / float64(NanosPerUnit)
+			opString += _indt(indentationCounter) + "Unstaking Limit: " +
+				strconv.FormatFloat(unstakeLimitDESO, 'f', 2, 64) + " $DESO\n"
+
+			indentationCounter--
+			opString += _indt(indentationCounter) + "]\n"
+			unstakeLimitStr = append(unstakeLimitStr, opString)
+		}
+		// Ensure deterministic ordering of the transaction count limit strings by doing a lexicographical sort.
+		sortStringsAndAddToLimitStr(unstakeLimitStr)
+		indentationCounter--
+	}
+
+	// UnlockStakeLimitMap
+	if len(tsl.UnlockStakeLimitMap) > 0 {
+		var unlockStakeLimitStr []string
+		str += _indt(indentationCounter) + "Unlocking Stake Restrictions:\n"
+		indentationCounter++
+		for limitKey, limit := range tsl.UnlockStakeLimitMap {
+			opString := _indt(indentationCounter) + "[\n"
+
+			indentationCounter++
+			// ValidatorPKID
+			validatorPublicKeyBase58Check := "Any"
+			if !limitKey.ValidatorPKID.Eq(&ZeroPKID) {
+				validatorPublicKeyBase58Check = Base58CheckEncode(limitKey.ValidatorPKID.ToBytes(), false, params)
+			}
+			opString += _indt(indentationCounter) + "Validator PKID: " + validatorPublicKeyBase58Check + "\n"
+			// StakerPKID
+			stakerPublicKeyBase58Check := Base58CheckEncode(limitKey.StakerPKID.ToBytes(), false, params)
+			opString += _indt(indentationCounter) + "Staker PKID: " + stakerPublicKeyBase58Check + "\n"
+			// UnlockStakeLimit
+			opString += _indt(indentationCounter) + "Transaction Count: " + strconv.FormatUint(limit, 10) + "\n"
+
+			indentationCounter--
+			opString += _indt(indentationCounter) + "]\n"
+			unlockStakeLimitStr = append(unlockStakeLimitStr, opString)
+		}
+		// Ensure deterministic ordering of the transaction count limit strings by doing a lexicographical sort.
+		sortStringsAndAddToLimitStr(unlockStakeLimitStr)
+		indentationCounter--
+	}
```

Comment:
I need to look through this piece again, but I think it looks good. Just a note to myself

Reply:
Sounds good. Also check out my test for the ToMetamaskStr() function which shows what the output would like for different stake/unstake/unlock stake derived key txn spending limit configurations.